### PR TITLE
Onboard TF 2.13 EC2/SM image to Autopatch

### PR DESCRIPTION
**Pull request by shantanutrip**
*GitHub Issue #, if available:*

**Note**: 
- If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

- All PR's are checked weekly for staleness. This PR will be closed if not updated in 30 days.

### Description
Onboard TF 2.13 EC2 image to Autopatch

### Tests run
**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [x] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

- TF2.13 Training/Inference EC2 [60a97f2](https://github.com/aws/deep-learning-containers/pull/3866/commits/60a97f2665d08f6afbecbfc64e980ed4c2c4d7ac)
- TF2.13 Training/Inference SM [2df3b10](https://github.com/aws/deep-learning-containers/pull/3866/commits/2df3b108831177152c21728d3d6f45d65a959ebc)

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**
<details>
<summary>Expand</summary>

- [ ] `sagemaker_remote_tests = true`
- [ ] `sagemaker_efa_tests = true`
- [ ] `sagemaker_rc_tests = true`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

</details>

### Formatting
- [ ] I have run `black -l 100` on my code (formatting tool: https://black.readthedocs.io/en/stable/getting_started.html)

#### Builds to Execute
<details>
<summary>Expand</summary>

Click the checkbox to enable a build to execute upon merge.

*Note: By default, pipelines are set to "latest". Replace with major.minor framework version if you do not want "latest".*

- [ ] build_pytorch_training_latest
- [ ] build_pytorch_inference_latest
- [ ] build_tensorflow_training_latest
- [ ] build_tensorflow_inference_latest

</details>

### PR Checklist 
<details>
<summary>Expand</summary>

- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

#### NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ec2_benchmark_tests = true` or `sagemaker_benchmark_tests = true`
</details>

### Pytest Marker Checklist
<details>
<summary>Expand</summary>

- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type
</details>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -479,9 +479,12 @@ def derive_prod_image_uri_using_image_config_from_buildspec(
     :param new_account_id: str, Account ID of the prod repo
     :return: str, image_uri
     """
-    prod_repo = image_config.get(
-        "release_repository"
-    ) or derive_prod_repository_using_image_config_from_buildspec(
+    prod_repo = (
```
Comment (kace) [inline src/utils.py:482]: Just wanted to confirm, this example code is not specific to TF 2.13 correct?

File: tensorflow/training/buildspec-2-13-ec2.yml [hunk 2]
```diff
@@ -45,25 +55,27 @@ images:
...
-    tag_python_version: &TAG_PYTHON_VERSION py310
-    cuda_version: &CUDA_VERSION cu118
-    os_version: &OS_VERSION ubuntu20.04
-    tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *CUDA_VERSION,
-      "-", *OS_VERSION, "-ec2-example" ]
-    docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /example, /Dockerfile., *DEVICE_TYPE ]
-    enable_test_promotion: true
-    context:
-      <<: *TRAINING_CONTEXT
+  # BuildTensorflowExampleGpuCu118Py310TrainingDockerImage:
```
Comment (kace) [inline tensorflow/training/buildspec-2-13-ec2.yml:65]: We don't want to include the example images?

File: tensorflow/training/buildspec-2-13-sm.yml [hunk 3]
```diff
@@ -30,25 +35,26 @@ images:
...
-    os_version: &OS_VERSION ubuntu20.04
-    tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *CUDA_VERSION,
-                 "-", *OS_VERSION, "-sagemaker" ]
-    docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /, *CUDA_VERSION,
-                         /Dockerfile., *DEVICE_TYPE ]
-    target: sagemaker
-    enable_test_promotion: true
-    context:
-      <<: *TRAINING_CONTEXT
+  # BuildTensorflowSageMakerGpuPy310Cu118TrainingDockerImage:
```
Comment (kace) [inline tensorflow/training/buildspec-2-13-sm.yml:44]: We don't want to include the Cu118 images either?

**Review by kace — DISMISSED** [review]
Just a few clarifying questions about our intentions. 

The pipelines appear to be passing for both EC2 and SM runs.

File: tensorflow/training/buildspec-2-13-sm.yml (hunk 3 above)
Comment (shantanutrip) [reply tensorflow/training/buildspec-2-13-sm.yml]: Thanks for catching this - uncommented the image. Will allow it to run on the pipeline and evaluate the failures.

File: src/utils.py (hunk 1 above)
Comment (shantanutrip) [reply src/utils.py:482]: Yes, it is not specific to TF2.13.

**Review by kace — APPROVED** [review]
Ack on adding in cu118 images.

**Pull request merged**